### PR TITLE
exec: kill forkexec on abnormal websocket closure

### PR DESCRIPTION
**Pull request by brauner**
Signed-off-by: Christian Brauner <christian.brauner@ubuntu.com>

### CODE REVIEW

**Comment by brauner** [comment]
Closes #2964.

File: lxd/container_exec.go [hunk 1]
```diff
@@ -160,6 +160,16 @@ func (s *execWs) Do(op *operation) error {
 
 				if err != nil {
 					shared.LogDebugf("Got error getting next reader %s", err)
+					if er, ok := err.(*websocket.CloseError); ok && er.Code == websocket.CloseAbnormalClosure {
```
Comment (stgraber) [inline lxd/container_exec.go]: Can we split that one into a few more lines, would be a bit easier to read I think :)

Comment (brauner) [inline lxd/container_exec.go]: ;)

**Comment by stgraber** [comment]
Just a bit of nitpicking in-line. Also, can we add a test for this? Should be reasonably straightforward.

**Pull request merged**